libsolidity/NameAndTypeResolver.cpp

```diff
@@ -82,6 +82,7 @@ void NameAndTypeResolver::checkTypeRequirements(ContractDefinition& _contract)
 	for (ASTPointer<StructDefinition> const& structDef: _contract.getDefinedStructs())
 		structDef->checkValidityOfMembers();
 	_contract.checkTypeRequirements();
+	checkDuplicateFunctionDeclaration(_contract); // TODO:
```

Comment:
I think this should rather be done inside `_contract.checkTypeRequirements`.
We have kind of a two-stage process: References to declarations are resolved using the `DeclarationContainer`, but it is quite "dumb" and actually thrown away after that stage.